src/ThundraWrapper.ts

```diff
@@ -218,22 +253,60 @@ class ThundraWrapper {
         }
     }
 
-    startDebuggerProxyIfAvailable(): void {
+    async startDebuggerProxyIfAvailable() {
         if (this.debuggerProxy) {
             this.finishDebuggerProxyIfAvailable();
         }
-        if (this.spawn && this.inspector) {
+        if (this.fork && this.inspector) {
             try {
-                this.debuggerProxy =
-                    this.spawn(
-                        '/opt/socat',
-                        [
-                            'TCP:' + this.brokerHost + ':' + this.brokerPort,
-                            'TCP:localhost:' + this.debuggerPort + ',forever',
-                        ],
-                        {detached: true});
-                this.inspector.open(this.debuggerPort, 'localhost', true);
-                this.waitForDebugger();
+                this.debuggerProxy = this.fork(
+                    path.join(__dirname, DEBUG_BRIDGE_FILE_NAME),
+                    [],
+                    {
+                        detached: true,
+                        env: {
+                            BROKER_HOST: this.brokerHost,
+                            BROKER_PORT: this.brokerPort,
+                            SESSION_NAME: this.sessionName,
+                            SESSION_TIMEOUT: this.sessionTimeout,
+                            AUTH_TOKEN: this.authToken,
+                            DEBUGGER_PORT: this.debuggerPort,
+                            LOGS_ENABLED: this.debuggerLogsEnabled,
+                            BROKER_WS_PROTOCOL,
+                        },
+                    },
+                );
+                this.inspector.open(this.debuggerPort, 'localhost', false);
+
+                const waitForBrokerConnection = () => new Promise((resolve) => {
+                    this.debuggerProxy.once('message', (mes: any) => {
+                        if (mes === 'brokerConnect') {
+                            return resolve(false);
+                        }
+
+                        if (typeof mes === 'string') {
+                            const match = mes.match(BROKER_WS_HTTP_ERROR_PATTERN);
+                            if (match) {
+                                const errCode = Number(match[1]);
+                                const errMessage = BROKER_WS_HTTP_ERR_CODE_TO_MSG[errCode];
+
+                                if (errMessage) {
+                                    ThundraLogger.getInstance().error('Thundra Debugger: ' + errMessage);
```

Comment:
I think, even if there is no match for message, as we accept is as error, it should be logged. ıf it matches pattern, we log with pre-defined messages, otherwise, log with raw message